apps/api-server/src/routes/api/project.js

```diff
@@ -302,15 +302,23 @@ router.route('/:projectId') //(\\d+)
     req.pendingMessages = [{ key: `project-${project.id}-update`, value: 'event' }];
     if (req.body.url && req.body.url != project.url) req.pendingMessages.push({ key: `project-urls-update`, value: 'event' });
 
+    // Check if url contains protocol
+    if (req.body.url && /^https?:\/\//i.test(req.body.url)) {
+      return next(new Error('URL cannot contain protocol (http, https)'));
+    }
+
     // Update allowedDomains if creating a new site
     let updateBody = req.body;
     if((project?.config?.widgets?.allowedDomains || []).length === 0 && req?.body?.url){
+      let url = new URL(req.body.url);
+      let hostname = url.hostname;
```

Comment:
de `new URL(...)` werkt toch niet als er geen protocol is opgegeven? Ik bedoelde het ook anders, je moet bij het opslaan in de `allowedDomains` het protocol eraf halen, zodat het enkel `openstad.org` is (dus de `url.hostname` in dit geval)

Reply:
De check op regel 305 mag er dus uit, en er moet een check bij op regel 313 of er een protocol in de url zit, zo niet; voeg dan een `http://` toe zodat de `new URL(...)` in ieder geval goed werkt!